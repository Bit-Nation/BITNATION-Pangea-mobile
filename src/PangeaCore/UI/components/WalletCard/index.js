// @flow

import React from 'react';
import { Image, View } from 'react-native';
import { Button, Text } from 'native-base';
import styles from './styles';
import Images from 'pangea-common-reactnative/assets/AssetsImages';
import i18n from 'pangea-common/i18n';

type Props = {
  /**
   * @desc Name for the wallet
   */
  nameHeading: string,
  /**
   * @desc Balance of the wallet
   */
  balance: string,
  /**
   * @desc Image to be displayed
   */
  imagePath: string,
  /**
   * @desc Callback on send money button press
   */
  onSendPress: () => void,
  /**
   * @desc Callback on receive money button press
   */
  onReceivePress: () => void,
  /**
   * @desc Callback on transactions button press
   */
  onTransactionPress: () => void,
};

/**
 * @desc Component for rendering wallet details.
 * @return {React.Component} A component.
 */
const WalletCard = ({
  imagePath,
  nameHeading,
  balance,
  onSendPress,
  onReceivePress,
  onTransactionPress,
}: Props) => (
  <View style={styles.container}>
    <View style={styles.row}>
      <Image style={styles.icon} source={imagePath} resizeMode='contain' />

      <View style={styles.textColumn}>
        <View style={styles.spacer} />
        <Text style={styles.nameHeading}>{nameHeading}</Text>
        <Text style={styles.nameSubheading}>{balance}</Text>
        <View style={styles.spacer} />

        <View style={styles.buttonsContainer}>
          <Button onPress={onSendPress} style={styles.button}>
            <Text style={styles.buttonText}>{i18n.t('common.send')}</Text>
          </Button>
          <Button onPress={onReceivePress} style={styles.button}>
            <Text style={styles.buttonText}>{i18n.t('common.receive')}</Text>
          </Button>
          <Button
            onPress={onTransactionPress}
            style={styles.button}
          >
            <Text style={styles.buttonText}>{i18n.t('common.transactions')}</Text>
          </Button>
        </View>
      </View>
    </View>
  </View>
);

WalletCard.defaultProps = {
  imagePath: Images.ethereumLogo,
  nameHeading: i18n.t('common.ethereum'),
  balance: '0',
  onSendPress: () => null,
  onReceivePress: () => null,
  onTransactionPress: () => null,
};

export default WalletCard;
