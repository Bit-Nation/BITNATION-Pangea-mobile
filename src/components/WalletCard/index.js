// @flow

import React from 'react';
import {
  Text,
  Image,
  View,
} from 'react-native';

import styles from './styles';
import Images from '../../global/AssetsImages';
import Button from '../common/ArrowButton';
import i18n from '../../global/i18n';

type Props = {
  /**
   * @desc Name for the wallet
   */
  nameHeading: string,
  /**
   * @desc Balance of the wallet
   */
  balance: string,
  /**
   * @desc Image to be displayed
   */
  imagePath: string,
  /**
   * @desc Callback on send money button press
   */
  onSendPress: () => void,
  /**
   * @desc Callback on receive money button press
   */
  onReceivePress: () => void,
  /**
   * @desc Callback on transactions button press
   */
  onTransactionPress: () => void,
}

/**
 * @desc Component for rendering wallet details.
 * @return {React.Component} A component.
 */
const WalletCard = ({
  imagePath, nameHeading, balance, onSendPress, onReceivePress, onTransactionPress,
}: Props) => (
  <View style={styles.container}>

    <View style={styles.row}>
      <Image style={styles.icon} source={imagePath} resizeMode='contain' />

      <View style={styles.textColumn}>
        <View style={styles.spacer} />
        <Text style={styles.nameHeading}>{nameHeading}</Text>
        <Text style={styles.nameSubheading}>{balance}</Text>
        <View style={styles.spacer} />

        <View style={styles.buttonsContainer}>
          <Button title={i18n.t('common.send')} onPress={onSendPress} style={[styles.button, styles.leftButton]} />
          <Button title={i18n.t('common.receive')} onPress={onReceivePress} style={styles.button} />
          <Button title={i18n.t('common.transactions')} onPress={onTransactionPress} style={styles.button} />
          <View style={styles.spacer} />
        </View>

        <View style={styles.spacer} />
      </View>
    </View>

  </View>
);

WalletCard.defaultProps = {
  imagePath: Images.ethereumLogo,
  nameHeading: i18n.t('common.ethereum'),
  balance: '0',
  onSendPress: () => null,
  onReceivePress: () => null,
};

export default WalletCard;
