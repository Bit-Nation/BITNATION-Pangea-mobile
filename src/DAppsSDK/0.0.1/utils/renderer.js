// @flow

import React from 'react';
import components from '../components';

export type ComponentsJSON = { type: string, props: Object, children: Array<Object> };

/**
 * @desc Function to get element class or function from text representation, e.g. 'Text' means Text component.
 * @param {string} typeName String representation of the type.
 * @return {*} Class or function of component or null if type is unknown.
 */
export const getTypeElementFromText = (typeName: string) => components[typeName];

/**
 * @desc Validates props against list of valid names and filter only whitelisted ones.
 * Used to prevent DApp developer to pass something bad to components we provide.
 * @param {Object} props Props to validate.
 * @param {string[]} validProps Object containing arrays by keys `native`, `stateBased` and `callbacks` according to corresponding type of props.
 * @param {string} type String representation of component type.
 * @return {Object} Props filtered by groups.
 */
export const validateProps = (
  props: Object,
  validProps: { native: Array<string>, custom: Array<string>, callbacks: Array<string> },
  type: string,
) => {
  const nativeProps = {};
  const customProps = {};
  const callbackProps = {};
  Object.keys(props).forEach((propName) => {
    if (typeof propName !== 'string') {
      console.warn('Prop name is not a string');
      return;
    }

    if (validProps.native.includes(propName)) {
      // It's an allowed native prop.
      nativeProps[propName] = props[propName];
    } else if (validProps.custom.includes(propName)) {
      // It's an allowed custom prop.
      customProps[propName] = props[propName];
    } else if (propName.endsWith('ID') && validProps.callbacks.includes(propName.substring(0, propName.length - 2))) {
      // It's an allowed prop to perform a callback. That properties are marked by ID suffix.
      callbackProps[propName] = props[propName];
    } else {
      console.warn(`Prop ${propName} is not allowed on component ${type}`);
    }
  });

  return { nativeProps, customProps, callbackProps };
};

/**
 * @desc Renders JSON provided by DApp developer into react components.
 * @param {ComponentsJSON} json JSON object that represents components to render.
 * @param {?string} key Key of component (used for updates).
 * @param {Function} customPropsProvider Function that takes component class or function and props and returns new props if it's needed to add ones.
 * @param {*} parent Function or class of component that is parent of currently used or undefined for root.
 * @return {*} Tree of components that is ready to be render.
 */
export const renderJSON = (json: ComponentsJSON, key: ?string, customPropsProvider: (component: any, ownProps: Object) => Object, parent: any) => {
  const { type, children } = json;

  // This is the case when we have string literal inside children.
  // It's only allowed for Text.
  if (type === undefined && typeof (json) === 'string') {
    if (parent === components.Text) {
      return json;
    }

    console.warn(`String literal may only appear inside Text component. Found "${json}" outside.`);
    return null;
  }

  const component = components[type];

  if (component == null) {
    console.warn(`Trying to render unknown component type ${type}.`);
    return null;
  }

  if (component.validNativeProps != null && Array.isArray(component.validNativeProps) === false) {
    console.warn(`validNativeProps field of component ${type} should be an array of strings.`);
    return null;
  }
  if (component.customProps != null && Array.isArray(component.customProps) === false) {
    console.warn(`customProps field of component ${type} should be an array of strings.`);
    return null;
  }
  if (component.callbackProps != null && Array.isArray(component.callbackProps) === false) {
    console.warn(`callbackProps field of component ${type} should be an array of strings.`);
    return null;
  }

  const { props } = json;

  const { nativeProps = props, customProps = {}, callbackProps = {} } =
    validateProps(json.props, {
      native: component.validNativeProps || [],
      custom: component.customProps || [],
      callbacks: component.callbackProps || [],
    }, type);

  return React.createElement(
    component,
    {
      nativeProps,
      ...customProps,
      ...customPropsProvider(component, { ...nativeProps, ...customProps, ...callbackProps }),
      key,
    },
    children
      ? children.map((child, index) => renderJSON(child, `${index}`, customPropsProvider, component))
      : null,
  );
};
