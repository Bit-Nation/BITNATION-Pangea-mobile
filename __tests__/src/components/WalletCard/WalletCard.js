import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import WalletCard from '../../../../src/components/WalletCard';
import Images from '../../../../src/global/AssetsImages';

describe('WalletCard component tests', () => {
  test('Rendering', () => {
    const tree = renderer.create(<WalletCard
      imagePath={Images.walletIcon}
      nameHeading='Testing Wallet'
      balance='Testing Balance'
      messageText='Testing Message'
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Press on Send/Receive Buttons', () => {
    const onPress = jest.fn();
    const wrapper = shallow(<WalletCard
      imagePath={Images.ethereumLogo}
      onSendPress={() => onPress}
      onReceivePress={() => onPress}
      nameHeading='Testing Press'
      balance='Testing balance'
    />);
    expect(wrapper).toMatchSnapshot();
    const render = wrapper.dive();
    render.find('TouchableOpacity').forEach((child) => {
      child.simulate('press');
      expect(onPress).toHaveBeenCalledTimes(1);
    });
  });
});
