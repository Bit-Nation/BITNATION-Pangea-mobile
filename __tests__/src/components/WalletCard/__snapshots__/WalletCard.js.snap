// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletCard component tests Press on Send/Receive Buttons 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletCard
    balance="Testing balance"
    imagePath={1}
    nameHeading="Testing Press"
    onReceivePress={[Function]}
    onSendPress={[Function]}
    onTransactionPress={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "borderRadius": 15,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "marginRight": 5,
              "marginTop": 36,
              "width": 55,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-start",
              "marginLeft": 0,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#929497",
                "fontSize": 17,
                "fontWeight": "bold",
                "marginBottom": 4,
              }
            }
          >
            Testing Press
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFCE00",
                "fontSize": 30,
              }
            }
          >
            Testing balance
          </Text>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Button
              enabled={true}
              onPress={[Function]}
              style={
                Object {
                  "height": 36,
                }
              }
              title="Send"
            />
            <Button
              enabled={true}
              onPress={[Function]}
              style={
                Object {
                  "height": 36,
                }
              }
              title="Receive"
            />
            <Button
              enabled={true}
              onPress={[Function]}
              style={
                Object {
                  "height": 36,
                }
              }
              title="Transactions"
            />
          </View>
        </View>
      </View>,
      "style": Object {
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "flex": 1,
        "height": 130,
        "marginTop": 5,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "marginRight": 5,
                "marginTop": 36,
                "width": 55,
              }
            }
          />,
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-start",
                "marginLeft": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#929497",
                  "fontSize": 17,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                }
              }
            >
              Testing Press
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#FFCE00",
                  "fontSize": 30,
                }
              }
            >
              Testing balance
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Button
                enabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "height": 36,
                  }
                }
                title="Send"
              />
              <Button
                enabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "height": 36,
                  }
                }
                title="Receive"
              />
              <Button
                enabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "height": 36,
                  }
                }
                title="Transactions"
              />
            </View>
          </View>,
        ],
        "style": Object {
          "alignItems": "stretch",
          "backgroundColor": "white",
          "borderRadius": 15,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMode": "contain",
            "source": 1,
            "style": Object {
              "marginRight": 5,
              "marginTop": 36,
              "width": 55,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#929497",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "marginBottom": 4,
                  }
                }
              >
                Testing Press
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#FFCE00",
                    "fontSize": 30,
                  }
                }
              >
                Testing balance
              </Text>,
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />,
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Button
                  enabled={true}
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 36,
                    }
                  }
                  title="Send"
                />
                <Button
                  enabled={true}
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 36,
                    }
                  }
                  title="Receive"
                />
                <Button
                  enabled={true}
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 36,
                    }
                  }
                  title="Transactions"
                />
              </View>,
            ],
            "style": Object {
              "flex": 1,
              "justifyContent": "flex-start",
              "marginLeft": 0,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Object {
                  "flex": 1,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Testing Press",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#929497",
                  "fontSize": 17,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                },
              },
              "ref": null,
              "rendered": "Testing Press",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Testing balance",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#FFCE00",
                  "fontSize": 30,
                },
              },
              "ref": null,
              "rendered": "Testing balance",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Object {
                  "flex": 1,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Button
                    enabled={true}
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 36,
                      }
                    }
                    title="Send"
                  />,
                  <Button
                    enabled={true}
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 36,
                      }
                    }
                    title="Receive"
                  />,
                  <Button
                    enabled={true}
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 36,
                      }
                    }
                    title="Transactions"
                  />,
                ],
                "style": Object {
                  "flexDirection": "row",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "enabled": true,
                    "onPress": [Function],
                    "style": Object {
                      "height": 36,
                    },
                    "title": "Send",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "enabled": true,
                    "onPress": [Function],
                    "style": Object {
                      "height": 36,
                    },
                    "title": "Receive",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "enabled": true,
                    "onPress": [Function],
                    "style": Object {
                      "height": 36,
                    },
                    "title": "Transactions",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "alignItems": "stretch",
              "backgroundColor": "white",
              "borderRadius": 15,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "marginRight": 5,
                "marginTop": 36,
                "width": 55,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-start",
                "marginLeft": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#929497",
                  "fontSize": 17,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                }
              }
            >
              Testing Press
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#FFCE00",
                  "fontSize": 30,
                }
              }
            >
              Testing balance
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Button
                enabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "height": 36,
                  }
                }
                title="Send"
              />
              <Button
                enabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "height": 36,
                  }
                }
                title="Receive"
              />
              <Button
                enabled={true}
                onPress={[Function]}
                style={
                  Object {
                    "height": 36,
                  }
                }
                title="Transactions"
              />
            </View>
          </View>
        </View>,
        "style": Object {
          "alignItems": "stretch",
          "backgroundColor": "transparent",
          "flex": 1,
          "height": 130,
          "marginTop": 5,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "marginRight": 5,
                  "marginTop": 36,
                  "width": 55,
                }
              }
            />,
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "marginLeft": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#929497",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "marginBottom": 4,
                  }
                }
              >
                Testing Press
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#FFCE00",
                    "fontSize": 30,
                  }
                }
              >
                Testing balance
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Button
                  enabled={true}
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 36,
                    }
                  }
                  title="Send"
                />
                <Button
                  enabled={true}
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 36,
                    }
                  }
                  title="Receive"
                />
                <Button
                  enabled={true}
                  onPress={[Function]}
                  style={
                    Object {
                      "height": 36,
                    }
                  }
                  title="Transactions"
                />
              </View>
            </View>,
          ],
          "style": Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
            "borderRadius": 15,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMode": "contain",
              "source": 1,
              "style": Object {
                "marginRight": 5,
                "marginTop": 36,
                "width": 55,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                />,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#929497",
                      "fontSize": 17,
                      "fontWeight": "bold",
                      "marginBottom": 4,
                    }
                  }
                >
                  Testing Press
                </Text>,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#FFCE00",
                      "fontSize": 30,
                    }
                  }
                >
                  Testing balance
                </Text>,
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                />,
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Button
                    enabled={true}
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 36,
                      }
                    }
                    title="Send"
                  />
                  <Button
                    enabled={true}
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 36,
                      }
                    }
                    title="Receive"
                  />
                  <Button
                    enabled={true}
                    onPress={[Function]}
                    style={
                      Object {
                        "height": 36,
                      }
                    }
                    title="Transactions"
                  />
                </View>,
              ],
              "style": Object {
                "flex": 1,
                "justifyContent": "flex-start",
                "marginLeft": 0,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "style": Object {
                    "flex": 1,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Testing Press",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#929497",
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "marginBottom": 4,
                  },
                },
                "ref": null,
                "rendered": "Testing Press",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Testing balance",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#FFCE00",
                    "fontSize": 30,
                  },
                },
                "ref": null,
                "rendered": "Testing balance",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "style": Object {
                    "flex": 1,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Button
                      enabled={true}
                      onPress={[Function]}
                      style={
                        Object {
                          "height": 36,
                        }
                      }
                      title="Send"
                    />,
                    <Button
                      enabled={true}
                      onPress={[Function]}
                      style={
                        Object {
                          "height": 36,
                        }
                      }
                      title="Receive"
                    />,
                    <Button
                      enabled={true}
                      onPress={[Function]}
                      style={
                        Object {
                          "height": 36,
                        }
                      }
                      title="Transactions"
                    />,
                  ],
                  "style": Object {
                    "flexDirection": "row",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "enabled": true,
                      "onPress": [Function],
                      "style": Object {
                        "height": 36,
                      },
                      "title": "Send",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "enabled": true,
                      "onPress": [Function],
                      "style": Object {
                        "height": 36,
                      },
                      "title": "Receive",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "enabled": true,
                      "onPress": [Function],
                      "style": Object {
                        "height": 36,
                      },
                      "title": "Transactions",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WalletCard component tests Rendering 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "flex": 1,
      "height": 130,
      "marginTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "marginRight": 5,
          "marginTop": 36,
          "width": 55,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "marginLeft": 0,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#929497",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Testing Wallet
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFCE00",
            "fontSize": 30,
          }
        }
      >
        Testing Balance
      </Text>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 15,
                "height": 30,
                "justifyContent": "center",
              },
              Object {
                "height": 36,
              },
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginLeft": 18,
                "opacity": 1,
              }
            }
            testID="Touchable"
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#FF8B00",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Send
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 15,
                "height": 30,
                "justifyContent": "center",
              },
              Object {
                "height": 36,
              },
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginLeft": 18,
                "opacity": 1,
              }
            }
            testID="Touchable"
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#FF8B00",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Receive
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 15,
                "height": 30,
                "justifyContent": "center",
              },
              Object {
                "height": 36,
              },
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginLeft": 18,
                "opacity": 1,
              }
            }
            testID="Touchable"
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#FF8B00",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Transactions
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
